app/javascript/components/remove-generic-item-modal.jsx

```diff
@@ -213,13 +213,17 @@ class RemoveGenericItemModal extends React.Component {
 RemoveGenericItemModal.propTypes = {
   dispatch: PropTypes.func.isRequired,
   recordId: PropTypes.number,
+  closefunc: PropTypes.func,
   gridChecks: PropTypes.arrayOf(PropTypes.any),
   modalData: PropTypes.objectOf(PropTypes.any).isRequired,
 };
 
 RemoveGenericItemModal.defaultProps = {
   recordId: null,
   gridChecks: null,
+  closefunc: () => {
```

Comment:
@iv1111 This is one of the file that uses provider dialogs modal( similar to your import modal), these are the changes that needs to be done in your pr, instead of console.log("aaa"), you can write your own logic here, i just added these changes as an example to show you. i will remove once it is working for you.